src/overlays/actors/ovl_En_Test/z_en_test.c

```diff
@@ -133,126 +204,1885 @@ static ColliderQuadInit D_808645C8 = {
+        } else {
+            this->actor.speedXZ *= -0.8f;
+            newYaw = 0;
+        }
+
+        if (ABS(newYaw) > 0x4000) {
+            this->actor.speedXZ *= -0.8f;
+
+            if (this->actor.speedXZ < 0.0f) {
+                this->actor.speedXZ -= 0.5f;
+            } else {
+                this->actor.speedXZ += 0.5f;
+            }
+        }
+    }
+
+    this->actor.world.rot.y = this->actor.shape.rot.y + 0x3FFF;
+
+    if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+        checkDist = 200.0f;
+    }
+
+    if (this->actor.xzDistToPlayer <= (80.0f + checkDist)) {
+        Math_SmoothStepToF(&this->unk_7EC, -2.5f, 1.0f, 0.8f, 0.0f);
+    } else if (this->actor.xzDistToPlayer > (110.0f + checkDist)) {
+        Math_SmoothStepToF(&this->unk_7EC, 2.5f, 1.0f, 0.8f, 0.0f);
+    } else {
+        Math_SmoothStepToF(&this->unk_7EC, 0.0f, 1.0f, 6.65f, 0.0f);
+    }
+
+    if (this->unk_7EC != 0.0f) {
+        this->actor.world.pos.x += (Math_SinS(this->actor.shape.rot.y) * this->unk_7EC);
+        this->actor.world.pos.z += (Math_CosS(this->actor.shape.rot.y) * this->unk_7EC);
+    }
+
+    this->skelAnime.playSpeed = this->actor.speedXZ * 0.5f;
+    prevFrame = this->skelAnime.curFrame;
+    SkelAnime_Update(&this->skelAnime);
+
+    temp_f16 = this->skelAnime.curFrame - ABS(this->skelAnime.playSpeed);
+    absPlaySpeed = ABS(this->skelAnime.playSpeed);
+
+    if ((this->timer % 32) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_WARAU);
+    }
+    if ((s32)this->skelAnime.curFrame != prevFrame) {
+        temp_v0_2 = (s32)absPlaySpeed + prevFrame;
+
+        if (((temp_v0_2 > 1) && (temp_f16 <= 0)) || ((temp_f16 < 7) && (temp_v0_2 >= 8))) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_WALK);
+        }
+    }
+
+    if (this->timer == 0) {
+        if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+            EnTest_SetupIdle(this);
+        } else if (Actor_IsTargeted(globalCtx, &this->actor)) {
+            if (!EnTest_ReactToProjectile(globalCtx, this)) {
+                EnTest_ChooseAction(this, globalCtx);
+            }
+        } else if (player->heldItemActionParam != PLAYER_AP_NONE) {
+            if ((globalCtx->gameplayFrames % 2) != 0) {
+                EnTest_SetupIdle(this);
+            } else {
+                EnTest_SetupWalkAndBlock(this);
+            }
+        } else {
+            EnTest_SetupWalkAndBlock(this);
+        }
+
+    } else {
+        this->timer--;
+    }
+}
+
+void func_80862DBC(EnTest* this, GlobalContext* globalCtx) {
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_DAMAGE);
+    this->unk_7C8 = 2;
+    BodyBreak_Alloc(&this->bodyBreak, 60, globalCtx);
+    this->actor.home.rot.x = 0;
+
+    if (this->swordState >= STALFOS_SWORD_OFF) {
+        EffectBlure_AddSpace(Effect_GetByIndex(this->effectIndex));
+        this->swordState = STALFOS_SWORD_NOBLUR;
+    }
+
+    this->actor.flags &= ~1;
+
+    if (this->actor.params == 5) {
```

Comment:
In your `StalfosType` enum you go up to type 3, did you miss this? or this is not related to the Stalfos' type?